teuthology/orchestra/connection.py

```diff
@@ -36,7 +37,12 @@ def create_key(keytype, key):
     elif keytype == 'ssh-ed25519':
         return paramiko.ed25519key.Ed25519Key(data=base64.decodestring(key.encode()))
     else:
-        raise ValueError('keytype must be ssh-rsa, ssh-dss (DSA) or ssh-ed25519')
+        l = "{hostname} {keytype} {key}".format(hostname="x", keytype=keytype, key=key)
+
+        j = HostKeyEntry.from_line(l)
+        if j is None:
+            raise ValueError('invalid host keytype {keytype}'.format(keytype = keytype))
+        return j.key
```

Comment:
Could you please get rid of negative check, that would have better readability:
```
if j:
        return j.key
else:
        raise ValueError('invalid host keytype {keytype}'.format(keytype = keytype))
```

Reply:
I pushed a new version of this:
/1/ use HostKeyEntry.from_line for all cases.
/2/ use "assert" to avoid use of "None".